ml-agents/mlagents/trainers/lesson_controller.py

```diff
@@ -0,0 +1,122 @@
+import os
+import yaml
+import math
+
+from .exception import LessonControllerError
+
+import logging
+
+logger = logging.getLogger("mlagents.trainers")
+
+
+class LessonController(object):
+    def __init__(self, location):
+        """
+        Initializes a Curriculum object.
+        :param location: Path to yaml file defining reset configuration
+        """
+        self.measure = None
+        try:
+            with open(location) as data_file:
+                self.data = yaml.load(data_file)
+        except IOError:
+            raise LessonControllerError("The file {0} could not be found.".format(location))
+        except UnicodeDecodeError:
+            raise LessonControllerError("There was an error decoding {}".format(location))
+        self.smoothing_value = 0
+        self.check_keys(location)
+        self.measure = self.data["measure"]
```

Comment:
nit: I think some of this can be cleaned up - you probably don't need to keep `self.data` at all, just store these keys and data["signal_smoothing"]. That simplifies change_lesson() a bit.